lang/java/reef-io/src/main/java/org/apache/reef/io/network/group/impl/task/OperatorTopologyStructImpl.java

```diff
@@ -234,6 +277,24 @@ private void sendToNode(final byte[] data, final ReefNetworkGroupCommProtos.Grou
         node, remove}));
     return retVal;
   }
+  
+  private byte[][] receiveArrayFromNode(final NodeStruct node, final boolean remove) {
```

Comment:
Do we need these *Array* methods? Couldn't we have got by linearizing the 2d array with some scheme & meta info and re-use the existing methods?